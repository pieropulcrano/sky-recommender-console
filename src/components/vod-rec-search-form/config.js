import CellExpand from '../form/data-grid-table/cell-expand/CellExpand';
import { formatToHumanReadable } from '../../utils/date';

export const columns = [
  { field: 'title', headerName: 'Title', width: '200', renderCell: CellExpand },
  {
    field: 'startProgram',
    headerName: 'Start Date',
    type: 'dateTime',
    width: '150',
    valueFormatter: (params) =>
      params.value && formatToHumanReadable(params.value),
  },
  {
    field: 'endProgram',
    headerName: 'End Date',
    type: 'dateTime',
    width: '150',
    valueFormatter: (params) =>
      params.value && formatToHumanReadable(params.value),
  },
  {
    field: 'vodSummaryLong',
    headerName: 'Summary',
    width: '200',
    type: 'dateTime',
    renderCell: CellExpand,
  },
  {
    field: 'branding',
    headerName: 'Branding',
    width: '100',
    renderCell: CellExpand,
  },
  { field: 'thumbnailType', headerName: 'Thumbnail Type', width: '150' },
  {
    field: 'verticalImageUrl',
    headerName: 'Vertical Image Url',
    width: '150',
    renderCell: CellExpand,
  },
];

export const initialValues = {
  title: '',
  selectedEvent: null,
};
