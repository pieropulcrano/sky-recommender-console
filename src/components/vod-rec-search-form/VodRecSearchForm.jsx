import React from 'react';
import PropTypes from 'prop-types';
import { Formik, Form } from 'formik';
import Grid from '@mui/material/Grid';
import Button from '@mui/material/Button';
import VodEventSearchForm from '../vod-event-search-form/VodEventSearchForm';
import Marginer from '../marginer/Marginer';
import DataGridTable from '../form/data-grid-table/DataGridTable';
import {
  RecSearchFormWrapper,
  RigthButtonWrapper,
} from '../common/Common.styled';
import { validationSchema } from './validation';
import { initialValues, columns } from './config';

/**
 * Component to search and fill a slot of the vod recommendation form with a vod event.
 */

const VodRecSearchForm = ({
  onSubmit,
  onSearch,
  isSearching,
  searchResult,
}) => {
  const handleSubmit = (values) => onSubmit(values);
  return (
    <RecSearchFormWrapper>
      <VodEventSearchForm onSubmit={onSearch} />
      <Formik
        initialValues={initialValues}
        onSubmit={handleSubmit}
        validationSchema={validationSchema}
        enableReinitialize
      >
        <Form>
          <Marginer direction="horizontal" margin={10} />
          <Grid container spacing={2}>
            <Grid item xs={12}>
              <DataGridTable
                name="selectedEvent"
                columns={columns}
                loading={isSearching}
                rows={searchResult}
              />
            </Grid>
            <Grid item xs={12}>
              <RigthButtonWrapper>
                <Button type="submit" variant="contained" color="success">
                  Select
                </Button>
              </RigthButtonWrapper>
            </Grid>
          </Grid>
        </Form>
      </Formik>
    </RecSearchFormWrapper>
  );
};

VodRecSearchForm.defaultProps = {
  searchResult: [],
};

VodRecSearchForm.propTypes = {
  /**
   * Result of the research
   */
  searchResult: PropTypes.array,
  /**
   * The form is performing a search operation
   */
  isSearching: PropTypes.bool.isRequired,
  /**
   * Called when the user clicks on create / update button
   */
  onSubmit: PropTypes.func.isRequired,
  /**
   * Called when the user clicks on search button
   */
  onSearch: PropTypes.func.isRequired,
};

export default VodRecSearchForm;
