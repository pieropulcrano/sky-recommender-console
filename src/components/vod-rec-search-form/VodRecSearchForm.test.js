import React from 'react';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import VodRecSearchForm from './VodRecSearchForm';
import vodEventFixture from '../../../fixtures/vod-event';
import { initialValues } from './config';

describe('Vod Rec Search Form', () => {
  let props;
  let handleSubmit;
  const selectBtn = 'Select';

  beforeEach(() => {
    handleSubmit = jest.fn();
    props = {
      onSubmit: handleSubmit,
      onSearch: jest.fn(),
      isSearching: false,
      searchResult: [],
      removeToken: jest.fn(),
    };
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('should render', () => {
    render(<VodRecSearchForm {...props} />);
    expect(document.body.childNodes[0].children).toMatchSnapshot();
  });

  //integration test of dataGridTable
  it('should display result', () => {
    props.searchResult = vodEventFixture.items;

    render(<VodRecSearchForm {...props} />);
    expect(
      screen.queryByText(vodEventFixture.items[0].title),
    ).toBeInTheDocument();
  });

  describe('Should not submit', () => {
    it('if no value is checked', async () => {
      render(<VodRecSearchForm {...props} />);

      const selectButton = screen.queryByText(selectBtn);
      await waitFor(() => fireEvent.click(selectButton));

      expect(props.onSubmit).not.toHaveBeenCalled();
    });
  });

  describe('Should submit', () => {
    //integration test of dataGridTable
    it('if a value is checked', async () => {
      props.searchResult = vodEventFixture.items;

      render(<VodRecSearchForm {...props} />);

      const checkbox = screen.queryByRole('checkbox');
      await waitFor(() => fireEvent.click(checkbox));

      const selectButton = screen.queryByText(selectBtn);
      await waitFor(() => fireEvent.click(selectButton));

      expect(props.onSubmit).toHaveBeenCalled();
      expect(props.onSubmit).toHaveBeenCalledWith({
        ...props.searchResult[0].selectedEvent,
        selectedEvent: {
          ...initialValues.selectedEvent,
          ...props.searchResult[0],
        },
        title: '',
      });
    });
  });
});
