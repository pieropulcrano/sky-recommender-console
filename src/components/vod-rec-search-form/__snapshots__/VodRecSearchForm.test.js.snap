// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vod Rec Search Form should render 1`] = `
HTMLCollection [
  <div
    class="css-fjocwo"
  >
    <form
      action="#"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="mui-1"
              id="mui-1-label"
            >
              Title
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-1"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Title
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Search
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
    <form
      action="#"
    >
      <span
        class="css-yviemf"
        direction="horizontal"
        margin="10"
      />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="css-x8ilf6"
          >
            <div
              aria-colcount="8"
              aria-multiselectable="false"
              aria-rowcount="0"
              class="MuiDataGrid-root css-j1c84w-MuiDataGrid-root"
              role="grid"
            >
              <div>
                <div />
              </div>
              <div
                class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
              >
                <div
                  style="height: -56px; width: 0px; position: absolute; top: 56px;"
                >
                  <div
                    class="MuiDataGrid-overlay css-1lk5ll6-MuiDataGrid-overlay"
                  >
                    No rows
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnHeaders css-f3jnds-MuiDataGrid-columnHeaders"
                  style="min-height: 56px; max-height: 56px; line-height: 56px;"
                >
                  <div
                    aria-rowindex="1"
                    class="MuiDataGrid-columnHeadersInner MuiDataGrid-columnHeadersInner--scrollable css-1s0hp0k-MuiDataGrid-columnHeadersInner"
                    role="row"
                    style="transform: translate3d(0px, 0px, 0px);"
                  >
                    <div
                      aria-colindex="1"
                      class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--alignCenter MuiDataGrid-columnHeaderCheckbox"
                      data-field="__check__"
                      role="columnheader"
                      style="width: 50px; min-width: 50px; max-width: 50px;"
                      tabindex="0"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderDraggableContainer"
                        draggable="false"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitleContainer"
                        >
                          <div
                            class="MuiDataGrid-columnHeaderTitleContainerContent"
                          >
                            <span
                              class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiDataGrid-checkboxInput css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                            >
                              <input
                                aria-label="Select all rows"
                                class="PrivateSwitchBase-input css-1m9pwf3"
                                data-indeterminate="false"
                                tabindex="-1"
                                type="checkbox"
                              />
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                                data-testid="CheckBoxOutlineBlankIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                                />
                              </svg>
                              <span
                                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                              />
                            </span>
                          </div>
                        </div>
                      </div>
                      <div
                        class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                        style="min-height: 56px; opacity: 1;"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                          data-testid="SeparatorIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M11 19V5h2v14z"
                          />
                        </svg>
                      </div>
                    </div>
                    <div
                      aria-colindex="2"
                      class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
                      data-field="title"
                      role="columnheader"
                      style="width: 200px; min-width: 200px; max-width: 200px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderDraggableContainer"
                        draggable="false"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitleContainer"
                        >
                          <div
                            class="MuiDataGrid-columnHeaderTitleContainerContent"
                          >
                            <div
                              aria-label=""
                              class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                              data-mui-internal-clone-element="true"
                            >
                              Title
                            </div>
                          </div>
                          <div
                            class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                          >
                            <button
                              aria-label="Sort"
                              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                              tabindex="-1"
                              title="Sort"
                              type="button"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                                data-testid="ArrowUpwardIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                                />
                              </svg>
                              <span
                                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                              />
                            </button>
                          </div>
                        </div>
                        <div
                          class="MuiDataGrid-menuIcon"
                        >
                          <button
                            aria-controls="mui-7"
                            aria-haspopup="true"
                            aria-label="Menu"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                            id="mui-8"
                            tabindex="-1"
                            title="Menu"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                              data-testid="TripleDotsVerticalIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                              />
                            </svg>
                            <span
                              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                      </div>
                      <div
                        class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                        style="min-height: 56px; opacity: 1;"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                          data-testid="SeparatorIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M11 19V5h2v14z"
                          />
                        </svg>
                      </div>
                    </div>
                    <div
                      aria-colindex="3"
                      class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
                      data-field="startProgram"
                      role="columnheader"
                      style="width: 150px; min-width: 150px; max-width: 150px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderDraggableContainer"
                        draggable="false"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitleContainer"
                        >
                          <div
                            class="MuiDataGrid-columnHeaderTitleContainerContent"
                          >
                            <div
                              aria-label=""
                              class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                              data-mui-internal-clone-element="true"
                            >
                              Start Date
                            </div>
                          </div>
                          <div
                            class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                          >
                            <button
                              aria-label="Sort"
                              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                              tabindex="-1"
                              title="Sort"
                              type="button"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                                data-testid="ArrowUpwardIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                                />
                              </svg>
                              <span
                                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                              />
                            </button>
                          </div>
                        </div>
                        <div
                          class="MuiDataGrid-menuIcon"
                        >
                          <button
                            aria-controls="mui-10"
                            aria-haspopup="true"
                            aria-label="Menu"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                            id="mui-11"
                            tabindex="-1"
                            title="Menu"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                              data-testid="TripleDotsVerticalIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                              />
                            </svg>
                            <span
                              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                      </div>
                      <div
                        class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                        style="min-height: 56px; opacity: 1;"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                          data-testid="SeparatorIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M11 19V5h2v14z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  style="overflow: visible; height: 0px; width: 0px;"
                >
                  <div
                    class="MuiDataGrid-virtualScroller css-1pans1z-MuiDataGrid-virtualScroller"
                    style="width: 0px; height: auto; margin-top: 56px; overflow-x: hidden;"
                  >
                    <div
                      class="MuiDataGrid-virtualScrollerContent MuiDataGrid-virtualScrollerContent--overflowed css-jz7yqw-MuiDataGrid-virtualScrollerContent"
                      style="width: auto; height: 1px; min-height: auto;"
                    >
                      <div
                        class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
                        style="transform: translate3d(0px, undefinedpx, 0px);"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="Mui-resizeTriggers"
                >
                  <div
                    class="expand-trigger"
                  >
                    <div
                      style="width: 1px; height: 1px;"
                    />
                  </div>
                  <div
                    class="contract-trigger"
                  />
                </div>
              </div>
              <div>
                <div
                  class="MuiDataGrid-footerContainer css-17jjc08-MuiDataGrid-footerContainer"
                >
                  <div />
                  <div
                    class="MuiTablePagination-root css-rtrcn9-MuiTablePagination-root"
                  >
                    <div
                      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-11bfvty-MuiToolbar-root-MuiTablePagination-toolbar"
                    >
                      <div
                        class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
                      />
                      <p
                        class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                        id="mui-3"
                      >
                        Rows per page:
                      </p>
                      <div
                        class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
                        variant="standard"
                      >
                        <div
                          aria-expanded="false"
                          aria-haspopup="listbox"
                          aria-labelledby="mui-3 mui-2"
                          class="MuiTablePagination-select MuiSelect-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                          id="mui-2"
                          role="button"
                          tabindex="0"
                        >
                          25
                        </div>
                        <input
                          aria-hidden="true"
                          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                          tabindex="-1"
                          value="25"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTablePagination-selectIcon MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                          data-testid="ArrowDropDownIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M7 10l5 5 5-5z"
                          />
                        </svg>
                      </div>
                      <p
                        class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
                      >
                        0–0 of 0
                      </p>
                      <div
                        class="MuiTablePagination-actions"
                      >
                        <button
                          aria-label="Go to previous page"
                          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                          disabled=""
                          tabindex="-1"
                          title="Go to previous page"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="KeyboardArrowLeftIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                            />
                          </svg>
                        </button>
                        <button
                          aria-label="Go to next page"
                          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                          disabled=""
                          tabindex="-1"
                          title="Go to next page"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="KeyboardArrowRightIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                            />
                          </svg>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="css-1diyz6t"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-ke5b6m-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Select
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>,
]
`;
